src/lib/supported_formats.js

```diff
@@ -1,17 +1,29 @@
-const FORMATS = {
+const STANDARD_FORMATS = {
   ogg: 'video/ogg; codecs="theora"',
   h264: 'video/mp4; codecs="avc1.42E01E"',
   webm: 'video/webm; codecs="vp8, vorbis"',
   vp9: 'video/webm; codecs="vp9"',
-  hls: 'application/x-mpegURL; codecs="avc1.42E01E"',
 };
 
+const HLS_FORMATS = {
+  hls: 'application/vnd.apple.mpegurl',
+  hlsLegacy: 'application/x-mpegURL',
+};
+
+export function getHlsFormats(videos) {
+  return videos.find(v => hlsFormats(v));
+}
+
+function hlsFormats(video) {
+  return Object.values(HLS_FORMATS).indexOf(video.mimeType()) != -1;
+}
+
 export function supportedFormats({ doc } = { doc: document }) {
   const v = doc.createElement('video');
   let supported = {};
-  for (const name in FORMATS) {
-    if (v.canPlayType(FORMATS[name]) === 'probably') {
-      supported[name] = FORMATS[name];
+  for (const name in STANDARD_FORMATS) {
+    if (v.canPlayType(STANDARD_FORMATS[name]) === 'probably') {
```

Comment:
I've not seen this code before, so "can you play this video?", "eh, _probably_ 🤷  ...", is making me laugh 😆 .